### ISSUE TITLE
[feature Request] Support for row with less columns

### ISSUE THREAD

**Issue by ashroyxi** [DenverCoder1/table2ascii#59]
if first row has 3 all rows have to have 3 columns. Is it possible to add a new row with 2 columns? currently not. in future?
![image](https://user-images.githubusercontent.com/72933395/205472283-5203e0e0-992e-4db2-9b15-7feea33487c9.png)

Here the last row, has two columns: Description and <bla bla bla>
i edited the text to show.

**Comment by DenverCoder1** [comment]
It's not so clear what the default behavior should be when the number of columns does not match.

There should ideally be an API for choosing which columns get merged in each row since in other cases you may want different cells to be merged into one, and not always just all the cells until the end of the row.

It's hard to come up with a very customizable interface that's also simple to use.

One solution could be to have a unique value to represent merging with the cell on the left, eg.

```py
table2ascii(
    body=[
		[1, 2, 3, 4, 5],
		["A", "Long cell value", MERGE_LEFT, MERGE_LEFT, MERGE_LEFT ]
	]
)
```

I'm not sure how common the use case is, let me know what your thoughts are on how it should be implemented for the user.

**Comment by ashroyxi** [comment]
I think your example case is pretty neat.

**Comment by DenverCoder1** [comment]
This is turning out to be much more complicated than anticipated for certain cases.

I've created a branch at https://github.com/DenverCoder1/table2ascii/tree/merge with what I've tried out.

You should be able to test it with `pip install -U git+https://github.com/DenverCoder1/table2ascii.git@merge`

For your example, it should be working just fine.

```py
from table2ascii import table2ascii
from table2ascii.merge import Merge
from table2ascii.alignment import Alignment
from table2ascii.preset_style import PresetStyle

output = table2ascii(
    header=["Name", "Price", "Category", "Stock", "Sku"],
    body=[
        ["test", 443, "test", 67, "test"],
    ],
    footer=["Description", "Long cell value that is merged", Merge.LEFT, Merge.LEFT, Merge.LEFT],
    alignments=[Alignment.LEFT] * 5,
    style=PresetStyle.thin_thick_rounded,
)

print(output)
```
![image](https://user-images.githubusercontent.com/20955511/206938201-cb211b14-8260-48e1-8522-e17a1cd1fbb7.png)

The cases where it still needs work include:

1. Using a "box" style - the column separators should be flat against merged cells

```py
output = table2ascii(
    body=[
        [1, 2, 3, 4, 5],
        ["A", "Long cell", Merge.LEFT, Merge.LEFT, Merge.LEFT ],
        ["Long cell value", Merge.LEFT, Merge.LEFT, Merge.LEFT, Merge.LEFT ],
    ],
    footer=["A", "B", "C", Merge.LEFT, "D"],
    alignments=[Alignment.LEFT, Alignment.RIGHT, Alignment.CENTER, Alignment.LEFT, Alignment.LEFT],
    style=PresetStyle.thin_box,
)
```

Current behavior:

![image](https://user-images.githubusercontent.com/20955511/206938493-68bde1c6-8461-4b6f-bb5f-c6d950c40f2a.png)

Preferred behavior:

![image](https://user-images.githubusercontent.com/20955511/206938509-2598438a-99ad-4d65-93e7-00afc9e0eaa8.png)
 
2. Cases where the merged cell text is longer than the combination of content in the cells from other rows - the long text goes past the end of the table

```py
output = table2ascii(
    body=[
        ["a", "b", "c", "d"],
        ["e", "Long cell value", Merge.LEFT, Merge.LEFT],
    ],
)
```

Current behavior:

![image](https://user-images.githubusercontent.com/20955511/206938621-39d48f23-2a44-4bbe-aec1-d3a8bbbbe245.png)

Preferred behavior:

![image](https://user-images.githubusercontent.com/20955511/206938638-d62c67f6-cdd5-4eb2-a9d0-651f606c9e95.png)

**Comment by DenverCoder1** [comment]
This feature is now released in version 1.0.1

**Comment by ashroyxi** [comment]
Thanks, This may be the only one program in this category which allows this. Thanks again for considering my request. I honor you.
